IPython/lib/inputhook.py

```diff
@@ -177,13 +197,17 @@ def clear_app_refs(self, gui=None):
             the app for that toolkit.  References are not held for gtk or tk
             as those toolkits don't have the notion of an app.
         """
+        warn("`clear_app_refs` is deprecated since IPython 5.0 and will be removed in future versions.",
```

Comment:
Yep that's annoying for tests. I did add the warning on import/instantiation, and removed it.